shared/lib/themes/actions.js

```diff
@@ -168,6 +168,19 @@ export function clearActivated() {
 	};
 };
 
+export function signup( theme ) {
+	return dispatch => {
+		const signupUrl = ThemeHelpers.getSignupUrl( theme );
+
+		dispatch( {
+			type: ThemeConstants.SIGNUP_WITH_THEME,
+			theme
+		} );
+
+		ThemeHelpers.navigateTo( signupUrl, false );
```

Comment:
`navigateTo` seems to use `page()` here, which messes with `pushState`. I don't think we want this, since we're navigating away from Calypso, so perhaps use a `window.location` or equivalent helper?